draft-ietf-quic-transport.md

```diff
@@ -1069,6 +1069,15 @@ to cease using the connection IDs when requested can result in connection
 failures, as the issuing endpoint might be unable to continue using the
 connection IDs with the active connection.
 
+An endpoint SHOULD limit the number of in flight RETIRE_CONNECTION_ID frames
+to bound the necessary state. In order to minimize delay in common situations,
+the limit on the number of in flight RETIRE_CONNECTION_IDs SHOULD be at least
+the active_connection_id_limit. An endpoint MAY choose to treat having too many
+connection IDs in need of retirement as a connection error of type
+CONNECTION_ID_LIMIT_ERROR.
```

Comment:
```suggestion
When retiring a connection ID, an endpoint remembers a sequence number and
maybe a packet number for a RETIRE_CONNECTION_ID frame.  An endpoint SHOULD
limit the state it commits to remembering connection IDs that need to be
retired to at least twice the active_connection_id_limit. An endpoint MAY
choose to treat having too many connection IDs in need of retirement as a
connection error of type CONNECTION_ID_LIMIT_ERROR.
```

Reply:
This suggestion looks good to me. I think we might want to clarify why the recommendation is at least twice (e.g., so that endpoints can track retirements it issued and those requested by the peer even under packet loss), but that's merely an editorial preference.

Reply:
I have received feedback from the list and the editors that "the lag" is the way to go. Unless that consensus is reversed, I believe that @martinthomson 's suggestion is too easily misinterpreted as allowing "the leak".  While I believe this can be fixed, I'll make a new suggestion that might be a better basis to making progress on what now is an editorial problem.

Reply:
```suggestion
When retiring a connection ID, an endpoint remembers a sequence number and maybe
a packet number for a RETIRE_CONNECTION_ID frame.  An endpoint SHOULD limit the
state it commits by limiting the number of connection IDs that it tracks for
retirement to at least twice the active_connection_id_limit. An endpoint MUST
NOT forget a connection ID without retiring it, though MAY choose to treat
having too many connection IDs in need of retirement as a connection error of
type CONNECTION_ID_LIMIT_ERROR.
```

I think that this is better than including a lot more works.  I've tweaked the last sentence to include a prohibition on forgetting connection IDs.

Reply:
I'm fine with @martinthomson's proposed text. 

It gives a recommendation, and also provides a dedicated error code to be used in such case. When a well-behaving endpoint receives that error code, it can blame that the peer is doing something wrong.

I think that this is the minimal fix that we need to apply, and I think that it is a reasonable way of making progress at late stage. If we are to apply a proper fix, my preference still goes to adopting #3553. It removes the need for "lagging," therefore, provides us having less complexity in the long run, at the cost of requiring a wire-format change.